src/inmanta/deploy/timers.py

```diff
@@ -47,56 +54,39 @@ def __init__(self, resource: ResourceIdStr, scheduler: "inmanta.deploy.scheduler
 
         self._resource_scheduler = scheduler
 
-    def ensure_timer(
+    def set_timer(
         self,
-        periodic_deploy_interval: int | None,
-        periodic_repair_interval: int | None,
-        is_dirty: bool,
+        countdown: int,
+        reason: str,
+        priority: TaskPriority,
     ) -> None:
         """
-        Make sure the underlying resource is marked for execution
-
-        :param periodic_deploy_interval: Per-resource deploy interval, assumed to be a positive int or None
-        :param periodic_repair_interval: Per-resource repair interval, assumed to be a positive int or None
-        :param is_dirty: Last known state of the resource at the timer install request time.
+        Main interface expected to be called once per timer by the TimerManager instance sitting on
+        top of this class.
+
+        Schedule the underlying resource for execution in <countdown> seconds with
+        the given reason and priority.
+
+        :param countdown: In this many seconds in the future, this method will ensure the underlying
+            resource is scheduled for deploy.
+        :param reason: The reason argument that will be passed down
+            to the resource scheduler's `deploy_with_context` method.
+        :param priority: The priority argument that will be passed down
+            to the resource scheduler's `deploy_with_context` method.
         """
 
-        next_execute_time: int
-
-        if periodic_repair_interval is None:
-            periodic_repair_interval = sys.maxsize
-        if periodic_deploy_interval is None:
-            periodic_deploy_interval = sys.maxsize
-
-        action_function: Callable[[ResourceIdStr, int], Coroutine[Any, Any, None]]
-        if is_dirty:
-            next_execute_time, action_function = min(
-                (periodic_deploy_interval, self._trigger_deploy_for_resource),
-                (periodic_repair_interval, self._trigger_repair_for_resource),
-                key=lambda x: x[0],
+        def _create_repair_task() -> None:
+            self.trigger_deploy = asyncio.create_task(
+                self._resource_scheduler.trigger_deploy_for_resource(self.resource, reason, priority)
             )
-        else:
-            next_execute_time = periodic_repair_interval
-            action_function = self._trigger_repair_for_resource
-
-        if next_execute_time == sys.maxsize:
-            # Per-resource timers are disabled
-            return
-
-        if next_execute_time == self.time_to_next_deploy:
-            # This timer is already in place
-            return
-
-        # Cancel previous schedule and re-schedule
-        self.cancel_next_schedule()
 
-        def _create_repair_task() -> None:
-            self.trigger_deploy = asyncio.create_task(action_function(self.resource, next_execute_time))
+        assert (
+            self.next_schedule_handle is None
```

Comment:
We don't seem to set it to `None` when we cancel the task.

Reply:
fixed